### PR TITLE
Add DB settings dialog (and store them)

### PR DESCRIPTION
**Pull request by kalbermattenm**
To be tested by @voisardf

### CODE REVIEW

File: Historisation/widgets/configuration/ajoutConfigurationBD_dialog.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        settings = QSettings()
+        is_defined = QSettings().value("HistorisationSITN/DB", None)
+        if is_defined:
+           self.pg_db_database.setText(settings.value("HistorisationSITN/DB", ''))
+           self.pg_db_host.setText(settings.value("HistorisationSITN/DBHost", ''))
+           self.pg_db_port.setText(settings.value("HistorisationSITN/DBPort", 5432))
+           self.pg_db_user.setText(settings.value("HistorisationSITN/DBUser", ''))
+           self.pg_db_password.setText(settings.value("HistorisationSITN/DBPassword", ''))
+    
+        self.pg_db_database.setText('toto2')
```
Comment (voisardf) [inline Historisation/widgets/configuration/ajoutConfigurationBD_dialog.py]: @kalbermattenm : found it 😉

Comment (kalbermattenm) [reply Historisation/widgets/configuration/ajoutConfigurationBD_dialog.py]: Thanks!!

File: Historisation/layer.py [hunk 2]
```diff
@@ -4,14 +4,21 @@
...
-
+from qgis.PyQt.QtCore import QSettings
 import psycopg2
 import re
 
 
 class Layer(object):
     def __init__(self, uri: QgsDataSourceUri):
         self.uri = uri
+        settings = QSettings()
```
Comment (maltaesousa) [inline Historisation/layer.py:15]: Nice!

File: Historisation/main.py [hunk 3]
```diff
@@ -151,14 +152,28 @@ def initGui(self):
         Create the menu entries and toolbar icons inside the QGIS GUI.
         '''
 
+        definedDB = QSettings().value("HistorisationSITN/DB", None)
+        enable_button = False
+        if definedDB is not None and definedDB != '':
```
Comment (maltaesousa) [inline Historisation/main.py]:
```suggestion
        if definedDB:
```

File: Historisation/widgets/configuration/ajoutConfigurationBD_dialog.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+    def onButtonClicked(self):
+
+        pg_db_database = self.pg_db_database.text()
+        pg_db_user = self.pg_db_user.text()
+        pg_db_password = self.pg_db_password.text()
+        pg_db_port = self.pg_db_port.text()
+        pg_db_host = self.pg_db_host.text()
+        errors = False
+        if pg_db_host is None or pg_db_host == '':
```
Comment (maltaesousa) [inline Historisation/widgets/configuration/ajoutConfigurationBD_dialog.py]: same as above:
```suggestion
        if not pg_db_host:
```

File: Historisation/widgets/configuration/ajoutConfigurationBD_dialog.ui [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+  <property name="geometry">
+   <rect>
+    <x>0</x>
+    <y>0</y>
+    <width>382</width>
+    <height>322</height>
+   </rect>
+  </property>
+  <property name="windowTitle">
+   <string>SITN 4D -Historisation d'une table</string>
```
Comment (maltaesousa) [inline Historisation/widgets/configuration/ajoutConfigurationBD_dialog.ui]:
```suggestion
   <string>SITN 4D - Historisation d'une table</string>
```

**Review by maltaesousa — APPROVED** [review]
Nice! I like this is using QSettings now :) I haven't tested it

**Pull request merged**